src/storage/nats_kv.rs

```diff
@@ -0,0 +1,248 @@
+//! Storage engine backed by NATS Kv
+//!
+//! This storage engine enables storing any WADM related information using NATS Kv as a backend
+
+use async_trait::async_trait;
+
+use async_nats::{
+    Client as NatsClient,
+    jetstream::kv::Config as JetstreamKvConfig,
+    jetstream::kv::Store as JetstreamKvStore,
+};
+
+use derive_builder::Builder;
+use semver::{Version, Prerelease, BuildMetadata};
+use serde::{Deserialize, Serialize};
+use serde_json::{
+    from_slice,
+    to_vec,
+    Error as SerdeJsonError,
+};
+use serde::ser::StdError;
+
+use super::{
+    EngineMetadata,
+    LatticeStorage,
+    StorageEngineError,
+    StorageError,
+    StoreOptions,
+};
+
+use super::state::{
+    LatticeState,
+    LatticeStateBuilder,
+    LatticeStateBuilderError,
+};
+
+use crate::storage::{
+    Store,
+    StoreRead,
+    StoreWrite,
+    StoreReadWrite,
+    WithStateMetadata,
+};
+
+const STORAGE_ENGINE_NAME: &str = "nats_kv";
+const STORAGE_ENGINE_VERSION: Version = Version {
+    major: 0,
+    minor: 1,
+    patch: 0,
+    pre: Prerelease::EMPTY,
+    build: BuildMetadata::EMPTY,
+};
+
+////////////
+// Engine //
+////////////
+
+#[derive(Debug)]
+pub struct NatsKvStorageEngine {
+    /// Configuration for the NATS KV storage engine
+    config: NatsKvStorageConfig,
+
+    /// NATS client
+    client: NatsClient,
+}
+
+#[derive(Debug, Builder, Serialize, Deserialize)]
+pub struct NatsKvStorageConfig {
+    /// URL to the nats instance
+    pub nats_url: String,
+
+    /// Prefix to prepend to NATS bucket name
+    pub nats_prefix: Option<String>,
+}
+
+impl NatsKvStorageEngine {
+    #[allow(dead_code)]
+    pub async fn new(config: NatsKvStorageConfig) -> Result<NatsKvStorageEngine, StorageEngineError> {
+        // Connect to NATS
+        let client = async_nats::connect(&config.nats_url).await
```

Comment:
This should probably be upgraded to support actual authentication. This code will only connect to anonymously authenticated servers, whereas most other wasmCloud components that talk to NATS support authenticating with the JWT+seed combo

Reply:
Yup, will update `NatsKvStorageConfig`

- [ ] Update `NatsKvStorageConfig` to support authenticated client connections (i.e. JWT + seed)